provisioner: add winrm_use_ssl arg to provisioner (#160)
hashicorp/packer-plugin-ansible: docs-partials/provisioner/ansible/Config-not-required.mdx

  this option will be changed to default to `false` for SSH and WinRM
  connections where the provisioner has access to a host IP.

> - `winrm_use_ssl` (boolean) - Force WinRM to use HTTPS instead of HTTP.

Author: Agreed, for consistency we can adopt a similar convention for this attribute.
Let's rename it

  this option will be changed to default to `false` for SSH and WinRM
  connections where the provisioner has access to a host IP.

- `ansible_winrm_use_http` (bool) - Force WinRM to use HTTP instead of HTTPS.
  

>   Set this to false to force Ansible to use HTTP instead of HTTPS to communicate

Reviewer: ```suggestion
  Set this to true to force Ansible to use HTTP instead of HTTPS to communicate
```

  over WinRM to the destination host.
  

>   This is because Ansible uses the port as a heuristic to determine whether
>   to use HTTP or not, and in the current state, since Packer assigns a
>   random port to connect to WinRM, Ansible's heuristic fails to determine that
>   it should be using HTTP, even when the communicator is setup to use it.

Reviewer: ```suggestion
  Ansible uses the port as a heuristic to determine whether to use 
  HTTP or not. In the current state, Packer assigns a random port
   for connecting to WinRM and Ansible's heuristic fails to determine that
  it should be using HTTP, even when the communicator is setup to use it.
```
